doc/classes/@GlobalScope.xml

```diff
@@ -2981,7 +2981,7 @@
 			Editing the property prompts the user for restarting the editor.
 		</constant>
 		<constant name="PROPERTY_USAGE_SCRIPT_VARIABLE" value="4096" enum="PropertyUsageFlags" is_bitfield="true">
-			The property is a script variable which should be serialized and saved in the scene file.
+			The property is a script variable which should be serialized and saved in the scene file. [constant PROPERTY_USAGE_SCRIPT_VARIABLE] automatically enables serialization like [constant PROPERTY_USAGE_STORAGE], but it does not automatically make the property show up in the editor (it must be combined with [constant PROPERTY_USAGE_EDITOR] to achieve that effect). [constant PROPERTY_USAGE_SCRIPT_VARIABLE] can be used to distinguish between script variables and built-in variables (which don't have this usage flag). [constant PROPERTY_USAGE_SCRIPT_VARIABLE] is [b]not[/b] applied to variables that are created by overriding [method Object._get_property_list] in a script.
```

Comment:
I don't think serializing is actually the intended behavior. See:

* https://github.com/godotengine/godot-proposals/discussions/7457
* https://github.com/godotengine/godot/pull/81377

Only `VariantWriter` pairs `PROPERTY_USAGE_SCRIPT_VARIABLE` and `PROPERTY_USAGE_STORAGE`, `marshalls.cpp`, `ResourceFormatSaverBinaryInstance` `ResourceFormatSaverTextInstance`, `Node`, `Resource` and others do not pair them.

https://github.com/godotengine/godot/blob/0f5f3bc9546b46b2029fc8896dc859697f1eab97/core/variant/variant_parser.cpp#L2159-L2161

https://github.com/godotengine/godot/blob/0f5f3bc9546b46b2029fc8896dc859697f1eab97/core/io/marshalls.cpp#L1677-L1680

As far as I understand, this flag is intended to distinguish script variables from native ones or those added via `_get_property_list()`. I found the following uses of `PROPERTY_USAGE_SCRIPT_VARIABLE`:

https://github.com/godotengine/godot/blob/0f5f3bc9546b46b2029fc8896dc859697f1eab97/editor/editor_inspector.cpp#L3115-L3117

https://github.com/godotengine/godot/blob/0f5f3bc9546b46b2029fc8896dc859697f1eab97/editor/property_selector.cpp#L134-L136

https://github.com/godotengine/godot/blob/0f5f3bc9546b46b2029fc8896dc859697f1eab97/modules/gltf/editor/editor_scene_exporter_gltf_settings.cpp#L33